### PR TITLE
Allow commit and clear transactions by id, 8.0.x

### LINKED ISSUE #3909: Allow transaction commits per Id
## Is your feature request related to a problem? Please describe.
Assume a client-server CRUD with error handling and/or server-side validation. If you get aggregated transactions for two updates on two separate rows with one that produces a server error, it is much cleaner to ```commit``` just the one that succeeded while leaving the other in a pending state for further action to the end-user.

## Describe the solution you'd like
Currently the commit method accepts the data to commit to. Allow it to accept the transaction ID/s to commit as well.

## Describe alternatives you've considered
n/a

## Additional context
Part of usability testing with an IgxHierarchicalGrid + ASP.NET Core WebAPI application.

### PR DESCRIPTION
**Pull request by wnvko**
Add optional `id` parameter to `commit` and `clear` transactions' methods.
Providing `id` in `commit` should commit to data source only transactions for provided id. This will also clear the redo stack if there is any.
Providing `id` to `clear` method will remove all transactions for provided `id`. This will also clear the redo stack if any, as well as remove the state for provided `id` if any.

Closes #3909 
  

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [x] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code
 - [x] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [x] This PR includes general feature table updates in the root `README.MD`
 - [x] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -187,6 +187,7 @@ In order to turn them off, you need to pass an argument and set it to `false`
...
     - `getNextCell` returns `ICellPosition` which defines the next cell, according to the current position, that match specific criteria. You can pass callback function as a third parameter of `getPreviousCell` method
     - `getPreviousCell` returns `ICellPosition` which defines the previous cell, according to the current position, that match specific criteria. You can pass callback function as a third parameter of `getPreviousCell` method.
+- `IgxTransactionService` commit and delete transaction by id - `commit` and `clear` methods of `IgxTransactionService` accept now optional `id` parameter. If `id` is provided to `commit` method only transactions for provided `id`. When `id` is provided to `clear` method all transactions for provided `id` will be removed from transactions log. Additionally both `commit` and `clear` when `id` is provided will remove all the actions in undo stack related to the provided `id`.
```
Comment (damyanpetev) [inline CHANGELOG.md]: I'd actually describe it as "commit and delete transaction(s) by record id" to make it clear this affects everything for a given primary key.
"x and y method of **the** service.."
Also you can simplify the second part by going "When id is provided to commit/clear, all transactions for this id are applied to the data or discarded, respectively. Additionally, any remaining entries in the undo stack matching the id are removed"

File: projects/igniteui-angular/src/lib/grids/api.service.ts [hunk 2]
```diff
@@ -132,12 +132,6 @@ export class GridBaseAPIService <T extends IgxGridBaseComponent & IGridDataBinda
 
         cell.editValue = value;
 
-        const valueInTransactions = this.grid.transactions.getAggregatedValue(cell.id.rowID, true);
```
[the remark this answers is not in the document]
Comment (wnvko) [reply projects/igniteui-angular/src/lib/grids/api.service.ts]: this actually does nothing

File: projects/igniteui-angular/src/lib/services/transaction/igx-hierarchical-transaction.ts [hunk 3]
```diff
@@ -51,13 +51,24 @@ export class IgxHierarchicalTransactionService<T extends HierarchicalTransaction
...
-        if (childDataKey) {
-            DataUtil.mergeHierarchicalTransactions(data, this.getAggregatedChanges(true), childDataKey, primaryKey, true);
+    /**
+     * Applies all transactions over the provided data
+     * @param data Data source to update
+     * @param primaryKey Primary key of the hierarchical data
+     * @param childDataKey Kye of child data collection
+     * @param id Optional record id to commit transactions for
+     */
+    public commit(data: any[], primaryKey?: any, childDataKey?: any, id?: any): void {
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/services/transaction/igx-hierarchical-transaction.ts]: I realize this is not a change from this PR, but.. Not a great fan of the mismatched signature here 😐 
Might be worth opening a new issue to look into that, cuz this service is still marked as hidden/experiamental but the feature is live and the HGrid doesn't override the transactions property, so you get auto-complete for the base type.

Comment (wnvko) [reply projects/igniteui-angular/src/lib/services/transaction/igx-hierarchical-transaction.ts]: this is for new issue 😏

**Pull request merged**